package/MDAnalysis/coordinates/LAMMPS.py

```diff
@@ -552,16 +579,51 @@ def _read_next_timestep(self):
 
         indices = np.zeros(self.n_atoms, dtype=int)
 
-        f.readline()  # ITEM ATOMS etc
-        for i in range(self.n_atoms):
-            idx, _, xs, ys, zs = f.readline().split()
+        atomline = f.readline()  # ITEM ATOMS etc
+        attrs = atomline.split()[2:]  # attributes on coordinate line
+        col_ids = {attr: i for i, attr in enumerate(attrs)}  # column numbers
+        keys = col_ids.keys()
+        # check for ids and what type of coordinate convention
+        ids = "id" in keys
+        # coord_dict keys = convention keys = column data or False
+        coord_dict = {conv: False for conv in self._conventions[1:]}
+        coord_dict['unscaled'] = self._get_column(
+            ["x", "y", "z"], col_ids)  # unscaled
+        coord_dict['scaled'] = self._get_column(
+            ["xs", "ys", "zs"], col_ids)  # scaled
+        coord_dict["unwrapped"] = self._get_column(
+            ["xu", "yu", "zu"], col_ids)  # unwrapped
+        coord_dict["scaled_unwrapped"] = self._get_column(
+            ["xsu", "ysu", "zsu"], col_ids)  # scaled unwrapped
```

Comment:
Does this one-liner need to be a static method, accessible to users? And does `coord_dict` need the "un/scaled/_unwrapped" labels if it's None? How about using a for loop and making it clear what the column names are this function, like

```python
COORDTYPE_COLUMN_NAMES = {
	"unscaled": ["x", "y", "z"],
	"scaled": ["xs", "ys", "zs"],
	...,
}
for coordtype, column_names in COORDTYPE_COLUMN_NAMES.items():
	if all(axis in attrs for axis in column_names):
		coord_dict[coordtype] = [attrs[axis] for axis in column_names]
```

Reply:
It was inline, richard suggested a lambda and oliver suggested a staticmethod, although I may have misinterpreted. Happy to condense it down, but hopefully we can settle on this one.